lib/models/user-identity.js

```diff
@@ -55,6 +55,32 @@
        username: username,
        password: password,
      };
+
+     if (typeof options.profileToUserFields !== 'undefined') {
+       for (var key in options.profileToUserFields) {
+         var fieldName = options.profileToUserFields[key];
+
+         if (fieldName.search('.') !== -1) {
```

Comment:
Do you want to test if fieldName contains `.`? If so, your code is NOT working as `.` will match any char. Use one of the following:
- `fieldName.indexOf('.') !== -1`
- `fieldName.search(/\./) !== -1`
- `fieldName.search('\\.') !== -1`